### PR TITLE
feat: Add bumpVersion to gradle manager

### LINKED ISSUE #7214: Add bumpVersion configuration option to gradle manager
**What would you like Renovate to be able to do?**

I am hoping to have renovate bump the version number within build.gradle whenever a dependency is updated. There appears to be a configuration option called `bumpVersion` that I believe is what I am looking for, but it is currently only for npm.

https://docs.renovatebot.com/configuration-options/#bumpversion
> Currently this setting supports npm only, so raise a feature request if you have a use for it with other package managers

**Did you already have any implementation ideas?**

#1413 appears to be where the npm feature was originally added, I was planning to use this as a template and add the required customization to the gradle manager.

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

Currently a human is intervening and bumping the version manually in each renovate pr, this isn't a critical issue, but would be nice to automate away.

**Is this a feature you'd be interested in implementing yourself?**

I am willing to implement this feature myself, I was planning on using #1413 as a template.

### PR DESCRIPTION
**Pull request by EricLemieux**
Updates the gradle manager to work with the bumpVersion configuration option. 

Even though in gradle it is possible to have the project version be an object, this assumes that the version is only configured as a
string.

I made [this test repository](https://github.com/EricLemieux/renovate-gradle-sample/pulls) in which I was testing the changes, where it appears to be working as intended.

Open questions that I have, unsure if they are blockers
* The project version field is named packageJsonVersion, should that be renamed in both npm and gradle to something more generic?
* `mirror:x` is missing, due to not having the gradle report, should I be reading off the file system again?

Closes #7214

### CODE REVIEW

**Comment by EricLemieux** [comment]
Some builds are failing due to an error in `Installing dependencies` and `Upload coverage`, I don't think any of my changes would have caused this. Should I just re-run until it works?

**Comment by rarkins** [comment]
I'll merge this PR first: #7338, after that it seems you have some snapshots to update (`yarn jest -u`)

**Review by rarkins — CHANGES_REQUESTED** [review]
Rename `packageJsonVersion` to `packageFileVersion`

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
@rarkins needs your review / approval

File: docs/usage/configuration-options.md
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:183]:
```suggestion
Currently this setting supports `npm` and `gradle` only, so raise a feature request if you have a use for it with other package managers.
Its purpose is if you want Renovate to update the `version` field within your file's `package.json` any time it updates dependencies within.
Usually this is for automatic release purposes, so that you don't need to add another step after Renovate before you can release a new version.
```

This suggestion ensures this text conforms to the one sentence per line style.
This suggestion also fixes a small `it's` vs `its` problem.

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs de-conflicting of a couple of files

**Comment by rarkins** [comment]
@EricLemieux is it possible to split this PR into two or more stages for easier review?

1. Refactor only - functionality unchanged
2. Add new functionality

**Comment by EricLemieux** [comment]
Just to confirm what you mean by multiple stages, is that two prs, or two
commits in the same pr?

**Comment by rarkins** [comment]
Two PRs

**Comment by rarkins** [comment]
If still interested, please resubmit first with refactor and then with functionality changes PRs

**Pull request closed without merging**
